client/api/Evaluator.cs

```diff
@@ -61,26 +61,58 @@ public bool BoolVariation(string key, dto.Target target, bool defaultValue)
         {
             Variation variation = EvaluateVariation(key, target, FeatureConfigKind.Boolean);
             bool res;
-            return (variation != null && Boolean.TryParse(variation.Value, out res)) ? res : defaultValue;
+            if (variation != null && Boolean.TryParse(variation.Value, out res))
+            {
```

Comment:
Looks good. I'm just wondering, will `variation` be null for all sad path cases during evaluation as far as you're aware? Including if any exceptions are thrown during evaluation?

Reply:
Not sure, but thats out of scope here. We're just adding a log message to the existing code flow/logic that already exists in production.

There is an Evaluator bug scrub ticket that addresses any sad path/exception handling improvements (if any) so will look at this sort of stuff there

Reply:
Sure. The reason I ask is that, if you check the SDK Code wiki, the SDK should log the error that caused the default variation to be returned either 

a) here, in the variation function log message or
b) somewhere down the call stack


So I am just checking, if the errors are logged somewhere else? As in this log message, we don't get a handle on the original error.

Reply:
If they are logged down the call stack, then that's fine. Some SDKs don't make it easy to propagate the errors up, so I added that caveat to the wiki. It's just so users can actually see _what error_ caused the default to be returned :)

Reply:
Plus, if you are not sure if `variation` being null is the cause of errors during evaluation, then can you please elaborate on the testing scenarios you covered for this log message. We just need to make sure it will be logged for failed evaluations correctly, so the existing production code _may_ need to be modified as was the case in Python - that's not beyond the scope of this ticket Chris.

Reply:
> Sure. The reason I ask is that, if you check the SDK Code wiki, the SDK should log the error that caused the default variation to be returned either
> 
> a) here, in the variation function log message or
> b) somewhere down the call stack
> 
> So I am just checking, if the errors are logged somewhere else? As in this log message, we don't get a handle on the original error.

I see, I take your point - no there is no logging that happens at the lower levels, any point the evaluator fails just returns Null and its caught here and default value returned. I can look at adding lower level logging.

> Plus, if you are not sure if variation being null is the cause of errors during evaluation, then can you please elaborate on the testing scenarios you covered for this log message. We just need to make sure it will be logged for failed evaluations correctly, so the existing production code may need to be modified as was the case in Python - that's not beyond the scope of this ticket Chris.

Are you implying theres an issue/bug with this code that returns the default variation in production? can you link the bug ticket so I can take a look? As for testing, I asked at standup for ideas, and IIRC it was yourself who had suggested I "point to QA and use a flag that doesn't exist". And IIRC you didn't suggest too many other test cases. Please provide a list of test cases I should carry out to satisfy your ask 👍  Apologies if I missed the wiki page for this, as still figuring out SDK's I don't have all the info in my head 😄  Thanks!

Reply:
Understood Chris. Yeah for testing that was one of the failed evaluation scenarios that came to mind - if a flag is not found. But it wasn't great advice in hindsight, as even if a flag is not found, it's up to the SDK itself to either log that error or raise an exception - apologies!

Yeah if you are happy to add the error logs to the point that the evaluation failed, that will be fine. Once that's done I think customers will have the info they need - as now, they won't know the cause of the error. I'll get the PR approved once that's done.

No, I am not implying there is a bug in production. 😂

Reply:
> No, I am not implying there is a bug in production. 😂

Good, thought there was an old CFD kicking about I was oblivious too 😂 

Makes sense to me - better to give the user as much info as possible so they stay away from us! I'll scribble down some more thorough tests too as I go through the code now I know its more than just "flag wasn't found" etc.

Good spot BTW 👍

Reply:
> I'll scribble down some more thorough tests too as I go through the code now I know its more than just "flag wasn't found" etc.

If we just ensure that anytime an error occurs during evaluation and it is logged, that will be fine. You can trigger each error if you like, for example if a flag isn't found (if the SDK handles that and doesn't just fall over) but the point is to understand where the error handling at a lower level and ensure that each time it is logged.